sunspec/core/client.py

```diff
@@ -207,32 +209,35 @@ def read_points(self):
                             read_len = self.addr + self.len - addr
                         data += self.device.read(addr, read_len)
                 if data:
-                    # print 'data len = ', len(data)
+                    # print('data len = ', len(data))
                     data_len = len(data)/2
                     if data_len != self.len:
                         raise SunSpecClientError('Error reading model %s' % self.model_type)
 
                     #  for each repeating block
                     for block in self.blocks:
                         # scale factor points
-                        for pname, point in block.points_sf.iteritems():
+                        for pname, point in block.points_sf.items():
                             offset = int(point.addr) - int(self.addr)
                             if point.point_type.data_to is not None:
                                 byte_offset = offset * 2
-                                # print pname, point, offset, byte_offset, (byte_offset + (int(point.point_type.len) * 2)), point.point_type.len
+                                # print(pname, point, offset, byte_offset, (byte_offset + (int(point.point_type.len) * 2)), point.point_type.len)
                                 point.value_base = point.point_type.data_to(data[byte_offset:byte_offset + (int(point.point_type.len) * 2)])
                                 if not point.point_type.is_impl(point.value_base):
                                     point.value_base = None
                             else:
                                 raise SunSpecClientError('No data_to function set for %s : %s' % (pname, point.point_type))
 
                         # non-scale factor points
-                        for pname, point in block.points.iteritems():
+                        for pname, point in block.points.items():
                             offset = int(point.addr) - int(self.addr)
                             if point.point_type.data_to is not None:
                                 byte_offset = offset * 2
-                                # print pname, point, offset, byte_offset, (byte_offset + (int(point.point_type.len) * 2)), point.point_type.len
+                                # print(pname, point, offset, byte_offset, (byte_offset + (int(point.point_type.len) * 2)), point.point_type.len)
                                 point.value_base = point.point_type.data_to(data[byte_offset:byte_offset + (int(point.point_type.len) * 2)])
+                                if (type(point.value_base) == bytes and
+                                        sys.version_info > (3,)):
```

Comment:
I meant to avoid having `if`s scattered around the code.  They basically reimplement generic features from the compatibility libraries in specific cases in our code.